machine_test.go

```diff
@@ -620,6 +621,13 @@ func testCreateMachine(ctx context.Context, t *testing.T, m *Machine) {
 	}
 }
 
+func testGetFirecrackerVersion(ctx context.Context, t *testing.T, m *Machine) {
+	_, err := m.getFirecrackerVersion(ctx)
```

Comment:
@kzys for assertion purposes would we prefer `firecracker_version` in Makefile be an environment variable so we can read it? Or perhaps using some regex to grab it dynamically `firecracker --version`? I think I prefer the former.

Reply:
I just made a change so that testGetFirecrackerVersion is now as follows: 

`func testGetFirecrackerVersion(ctx context.Context, t *testing.T, m *Machine) {
	ver, err := m.GetFirecrackerVersion(ctx)

	if err != nil {
		t.Errorf("GetFirecrackerVersion: %v", err)
	}

	if ver != "1.0.0" {
		t.Errorf("GetFirecrackerVersion: Exppected version 1.0.0, got version %v", ver)
	}
}`

This runs fine on my end using Firecracker version 1.0.0, but when tested on here it fails, as you can see below, and I'm not entirely sure if I can get the expected firecracker version from the makefile. Any suggestions?

Reply:
A bit more info:

Testing this against (I assume) the code in the branch seems to run perfectly fine, with no errors.

However, the pipeline also tests on the mainline of Firecracker, and that's when I get the following error:

```
=== RUN   TestMicroVMExecution/TestGetFirecrackerVersion
    machine_test.go:632: GetFirecrackerVersion: Exppected version 1.0.0, got version 1.1.0-35-g0fa080b1-dirty<!--
```

Looks like the failing test is the one running against the tip of Firecracker instead of any specific one specified by the Makefile, which I guess scratches off the idea of taking the version from the Makefile.

Even more tempted now to re-remove `ver` and subsequent checks on it but I just want to double check with you guys before going through with it.

Reply:
@sondavidb how about we try the following:

1. Use `firecracker_version` as an environment variable when running the tests.
2. Read the expected value from the environment variable during test execution.
3. Create a test utility function for sanitizing the dirty version returned by firecracker. For reference see https://github.com/firecracker-microvm/firecracker/blob/b3cc331794013419726446f9eb42d6e64f65d135/tests/framework/utils.py#L641

Reply:
Spoke with @sondavidb offline on why the environment variable would prove challenging in pipeline steps were we run against firecracker main.

@sondavidb, how about the following approach to read the version from the currently installed firecracker binary. Should make for more robust testing.

```go
func parseVersionFromStdout(stdout []byte) (string, error) {
	pattern := regexp.MustCompile(`Firecracker v(?P<version>[0-9]\.[0-9]\.[0-9])`)
	matches := pattern.FindStringSubmatch(string(stdout))
	if len(matches) != 2 {
		return "", fmt.Errorf("Unable to parse firecracker version from stdout")
	}
	return matches[pattern.SubexpIndex("version")], nil
}

func getFirecrackerVersion() (string, error) {
	cmd := exec.Command("firecracker", "--version")
	stdout, err := cmd.Output()
	if err != nil {
		return "", err
	}
	return parseVersionFromStdout(stdout)
}
```